VelocidiSDK/VelocidiSDK/VSDKRequest.m

```diff
@@ -0,0 +1,71 @@
+@import Foundation;
+#import "VSDKRequest.h"
+#import "VSDKUtil.h"
+#import <AdSupport/ASIdentifierManager.h>
+
+@implementation VSDKRequest
+
++ (NSString *) getTrackingNotAllowed{ return @"VSDKTrackingNotAllowedError"; }
+- (VSDKUtil *) getUtil{ return [[VSDKUtil alloc] init]; }
+- (ASIdentifierManager *) getIdentifierManager{ return [ASIdentifierManager sharedManager]; }
+
+- (instancetype)initWithHTTPSessionManager:(AFHTTPSessionManager *)manager{
+    if (self = [self init]) {
+        _manager = manager;
+    }
+    return self;
+}
+
+- (instancetype)init {
+    if (self = [super init]) {
+        _manager = [AFHTTPSessionManager manager];
+        [VSDKUtil setAcceptAllResponses:_manager];
+    }
+    return self;
+}
+
+- (void)performRequest: (void (^)(NSURLResponse *response, id responseObject))onSuccessBlock
+        :(void (^)(NSError *error))onFailureBlock {
+    if (![self.identifierManager isAdvertisingTrackingEnabled]) {
+        NSDictionary *userInfo = @{
+                                   NSLocalizedDescriptionKey: NSLocalizedString(@"Operation cannot be completed. Tracking is not allowed", nil),
+                                   NSLocalizedFailureReasonErrorKey: NSLocalizedString(@"The user has opted-out of ad tracking (Limited Ad Tracking is enabled in the user's device)", nil)};
+        NSError * trackingNotAllowedError = [NSError errorWithDomain: VSDKRequest.trackingNotAllowedError code: 1 userInfo: userInfo];
+        return onFailureBlock(trackingNotAllowedError);
```

Comment:
This will always be the same error. Can't we reuse it between method calls?

Reply:
Fixed in fb6f6cb